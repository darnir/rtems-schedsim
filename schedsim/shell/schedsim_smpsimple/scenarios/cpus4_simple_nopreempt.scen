#
#  Verify no-preempt is honored
#
rtems_init 4
task_create -p smp1 5
task_create -p smp2 6
task_create -p smp3 7
task_create -p smp4 8
dispatch
# GOAL: Tasks 0x0a010001 to 0x0a010004 as heir and executing on
# GOAL:    CPUs 1 - 4, respectively.
cpus
# Tasks are created and distributed across CPUs
task_create smp5 1
# GOAL: Task 0x0a010005 is heir but 0x0a010004 is executing on CPU 4
cpus
# GOAL: Task 0x0a010005 is heir but 0x0a010004 is executing on CPU 4
cpus
