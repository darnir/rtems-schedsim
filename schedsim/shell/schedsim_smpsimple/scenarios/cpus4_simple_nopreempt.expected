Script File               : scenarios/cpus4_simple_nopreempt.scen
verbose                   : 1
==> 1: #
==> 2: #  Verify no-preempt is honored
==> 3: #
==> 4: rtems_init 4
  Thread Heir: 0x09010001 priority=255
  Thread Executing: 0x09010001 priority=255
==> 5: task_create -p smp1 5
Creating task with:  Preemption: no Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=5
Task (smp1) starting: id=0x0a010001, priority=5
  Thread Heir: 0x0a010001 priority=5
  Thread Executing: 0x0a010001 priority=5
==> 6: task_create -p smp2 6
Creating task with:  Preemption: no Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=6
Task (smp2) starting: id=0x0a010002, priority=6
==> 7: task_create -p smp3 7
Creating task with:  Preemption: no Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=7
Task (smp3) starting: id=0x0a010003, priority=7
==> 8: task_create -p smp4 8
Creating task with:  Preemption: no Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=8
Task (smp4) starting: id=0x0a010004, priority=8
==> 9: dispatch
=== Invoke Thread Dispatch on CPU 1
  Thread Heir: 0x0a010002 priority=6
  Thread Executing: 0x09010002 priority=255
  Thread Executing: 0x0a010002 priority=6
=== Invoke Thread Dispatch on CPU 2
  Thread Heir: 0x0a010003 priority=7
  Thread Executing: 0x09010003 priority=255
  Thread Executing: 0x0a010003 priority=7
=== Invoke Thread Dispatch on CPU 3
  Thread Heir: 0x0a010004 priority=8
  Thread Executing: 0x09010004 priority=255
  Thread Executing: 0x0a010004 priority=8
==> 10: # GOAL: Tasks 0x0a010001 to 0x0a010004 as heir and executing on
==> 11: # GOAL:    CPUs 1 - 4, respectively.
==> 12: cpus
=== CPU Status
            EXECUTING    /     HEIR         /    IDLE        /  SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5 / 0x09010001 @255         false
  CPU 1: 0x0a010002 @  6 / 0x0a010002 @  6 / 0x09010002 @255         false
  CPU 2: 0x0a010003 @  7 / 0x0a010003 @  7 / 0x09010003 @255         false
  CPU 3: 0x0a010004 @  8 / 0x0a010004 @  8 / 0x09010004 @255         false
=== End of Ready Set of Threads
==> 13: # Tasks are created and distributed across CPUs
==> 14: task_create smp5 1
Creating task with:  Preemption: yes Timeslicing: no
  Thread Heir: 0x0a010001 priority=5
  Thread Executing: 0x0a010001 priority=5
Task (smp5) created: id=0x0a010005, priority=1
Task (smp5) starting: id=0x0a010005, priority=1
==> 15: # GOAL: Task 0x0a010005 is heir but 0x0a010004 is executing on CPU 4
==> 16: cpus
=== CPU Status
            EXECUTING    /     HEIR         /    IDLE        /  SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5 / 0x09010001 @255         false
  CPU 1: 0x0a010002 @  6 / 0x0a010002 @  6 / 0x09010002 @255         false
  CPU 2: 0x0a010003 @  7 / 0x0a010003 @  7 / 0x09010003 @255         false
  CPU 3: 0x0a010004 @  8 / 0x0a010005 @  1 / 0x09010004 @255         false
=== End of Ready Set of Threads
==> 17: dispatch
==> 18: # GOAL: Task 0x0a010005 is heir but 0x0a010004 is executing on CPU 4
==> 19: cpus
=== CPU Status
            EXECUTING    /     HEIR         /    IDLE        /  SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5 / 0x09010001 @255         false
  CPU 1: 0x0a010002 @  6 / 0x0a010002 @  6 / 0x09010002 @255         false
  CPU 2: 0x0a010003 @  7 / 0x0a010003 @  7 / 0x09010003 @255         false
  CPU 3: 0x0a010004 @  8 / 0x0a010005 @  1 / 0x09010004 @255         false
=== End of Ready Set of Threads
